proposals/low-level-struct-improvements.md

```diff
@@ -805,25 +730,119 @@ For example the last line of calculating *safe-to-escape* of returns will change
 
 > the safe-to-escape of all argument expressions including the receiver. **This will exclude all arguments that line up with parameters marked as [DoesNotEscape]**
 
-Misc Notes:
-- The  `DoesNotEscapeAttribute` will be defined in the `System.Runtime.CompilerServices` namespace.
-- The `DoesNotEscapeAttribute` cannot be combined with the `[ThisRefEscapes]` attribute, doing so results in an error.
-- The `DoesNotEscapeAttribute` will be emitted as a `modreq`
+Detailed Notes:
+- The `[DoesNotEscape]` and `[RefThisEscapes]` cannot be combined on the same method 
+
+### Safe fixed size buffers
+The language will relax the restrictions on fixed sized arrays such that they can be declared in safe code and the element type can be managed or unmanaged.  This will make types like the following legal:
+
+```c#
+internal struct CharBuffer
+{
+    internal char Data[128];
+}
+```
+
+These declarations, much like their `unsafe` counter parts, will define a sequence of `N` elements in the containing type. These members can be accessed with an indexer and can also be converted to `Span<T>` and `ReadOnlySpan<T>` instances.
+
+When indexing into a `fixed` buffer of type `T` the `readonly` state of the container must be taken into account.  If the container is `readonly` then the indexer returns `ref readonly T` else it returns `ref T`. 
+
+Accessing a `fixed` buffer without an indexer has no natural type however it is convertible to `Span<T>` types. In the case the container is `readonly` the buffer is implicitly convertible to `ReadOnlySpan<T>`, else it can implicitly convert to `Span<T>` or `ReadOnlySpan<T>` (the `Span<T>` conversion is considered *better*). 
+
+The resulting `Span<T>` instance will have a length equal to the size declared on the `fixed` buffer. The *safe-to-escape* scope of the returned value will be equal to the *safe-to-escape* scope of the container, just as it would if the backing data was accessed as a field.
+
+For each `fixed` declaration in a type where the element type is `T` the language will generate a corresponding `get` only indexer method whose return type is `ref T`. The indexer will be annotated with the `[RefThisEscapes]` attribute as the implementation will be returning fields of the declaring type. The accessibility of the member will match the accessibility on the `fixed` field.
+
+For example, the signature of the indexer for `CharBuffer.Data` will be the following:
+
+```c#
+[RefThisEscapes]
+internal ref char <>DataIndexer(int index) => ...;
```

Comment:
Is the `<>` correct in the above signature?

Reply:
Yes it's meant to indicate that it's a compiler generated member that can't be directly accessed by consumer code. The `<>` is part of all synthesized non-acecssible members we generate (usually has a bit of text in between)